build: Make introspection optional in meson

mattst88 opened the pull request.

mattst88 commented:
I wrote this patch last week when packaging colord-1.4.5 for Gentoo, but now I cannot figure out how to reproduce the problem it fixed! :facepalm:

IIRC the issue was that some of the code that I protect with `if get_option('vapi')` used a variable that was only defined (in another file) if `get_option('vapi')` was true.

I'm going to remove the patch from Gentoo and see if anyone notices, and then update this PR.

hughsie commented:
We need the GIR even if we compile without Vala. The idea is that the vapi is generated from the gir.

mattst88 commented:
Okay, removed the patch and got reports that showed what problem I'd forgotten I'd solved: https://bugs.gentoo.org/786873

Basically, there's no way to disable introspection, so if you don't have it enabled in dependencies it's not possible to build colord (since it's not possible to disable it).

I've replaced my patch with one from another contributor (@dm0-) that adds a flag to meson to control introspection, like there was in autotools.
